bundle/src/main/java/com/adobe/acs/commons/workflow/process/impl/SetImageOrientationProcess.java

```diff
@@ -0,0 +1,255 @@
+import com.day.cq.dam.api.Asset;
+import com.day.cq.dam.api.DamConstants;
+import com.day.cq.dam.commons.util.DamUtil;
+import com.day.cq.tagging.Tag;
+import com.day.cq.tagging.TagManager;
+import com.day.cq.wcm.resource.details.AssetDetails;
+import org.apache.commons.lang.ArrayUtils;
+import org.apache.commons.lang.StringUtils;
+import org.apache.felix.scr.annotations.Component;
+import org.apache.felix.scr.annotations.Properties;
+import org.apache.felix.scr.annotations.Property;
+import org.apache.felix.scr.annotations.Service;
+import org.apache.felix.scr.annotations.Reference;
+import org.apache.jackrabbit.JcrConstants;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.jcr.RepositoryException;
+import java.util.ArrayList;
+import java.util.List;
+
+
+@Component(
+        metatype = true,
+        label = "ACS AEM Commons - Workflow Process - Set Image Orientation",
+        description = "Determine image orientation from width/height and update image tags"
+)
+@Properties({
+        @Property(
+                label = "Workflow Label",
+                name = "process.label",
+                value = "Set Image Orientation",
+                description = "Determine image orientation from width/height and update image tags"
+        )
+})
+@Service
+public class SetImageOrientationProcess implements WorkflowProcess {
+    private static final Logger log = LoggerFactory.getLogger(SetImageOrientationProcess.class);
+
+    @SuppressWarnings("unused")
+    @Reference
+    private WorkflowPackageManager workflowPackageManager;
+
+    @SuppressWarnings("unused")
+    @Reference
+    private WorkflowHelper workflowHelper;
+
+    private static final String DEFAULT_CONFIG = ">1.1 properties:orientation/landscape\r\n"
+            + "<0.9 properties:orientation/portrait\r\n"
+            + "default properties:orientation/square";
+
+
+    public final void execute(WorkItem workItem, WorkflowSession workflowSession, MetaDataMap metaDataMap) {
+
+        Configuration config = new Configuration(metaDataMap);
+
+        try {
+
+            ResourceResolver resourceResolver = workflowHelper.getResourceResolver(workflowSession);
+
+            TagManager tagManager = resourceResolver.adaptTo(TagManager.class);
+            if (tagManager == null) {
+                log.error("Unable to adapt to Tag Manager. This step can't do it's work");
+                return;
+            }
+
+            final List<String> payloads = workflowPackageManager.getPaths(resourceResolver, (String) workItem.getWorkflowData().getPayload());
+            for (final String payload : payloads) {
+
+                log.debug("Processing {}", payload);
+
+                Resource payloadResource = resourceResolver.resolve(payload);
+
+                Asset asset = DamUtil.resolveToAsset(payloadResource);
```

Comment:
Might want to null check this, i think `DamUtil.resolveToAsset(payloadResource)` can return null if its pointing to something that cant be resolved to an asset (such as a Page)

Reply:
Good point; improved NPE handling and added another log entry to warn about these.